central/graphql/resolvers/nodes.go

```diff
@@ -385,31 +439,61 @@ func (resolver *nodeResolver) Vulns(ctx context.Context, args PaginatedQuery) ([
 	}), PaginatedQuery{Query: &query, Pagination: args.Pagination})
 }
 
+// Vulnerabilities returns all of the vulnerabilities in the node.
+func (resolver *nodeResolver) Vulnerabilities(ctx context.Context, args PaginatedQuery) ([]NodeVulnerabilityResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Nodes, "Vulnerabilities")
+	if err := readNodes(ctx); err != nil {
+		return nil, err
+	}
+
+	if !features.PostgresDatastore.Enabled() {
+		query := search.AddRawQueriesAsConjunction(args.String(), resolver.getNodeRawQuery())
+
+		return resolver.root.NodeVulnerabilities(scoped.Context(ctx, scoped.Scope{
+			Level: v1.SearchCategory_NODES,
+			ID:    resolver.data.GetId(),
+		}), PaginatedQuery{Query: &query, Pagination: args.Pagination})
+	}
```

Comment:
This code gets repeated for each of the vulnerability specific resolvers and could be refactored out into a single function. If you would like an example of how that would look you can refer to the changes in my PR #1537.

Reply:
Sounds right. Created a separate function for scoping.